### PR TITLE
[WIP] Use the new session api

### PR DESCRIPTION
**Pull request by blink1073**
cf #236 

cc @minrk

### CODE REVIEW

**Comment by blink1073** [comment]
The updated integration tests pass.

File: src/session/default.ts [hunk 1]
```diff
@@ -211,13 +230,45 @@ class DefaultSession implements ISession {
...
+   *
+   * #### Notes
+   * This uses the Jupyter REST API, and the response is validated.
+   * The promise is fulfilled on a valid response and rejected otherwise.
+   */
+  setType(type: string): Promise<void> {
+    if (this.isDisposed) {
+      return Promise.reject(new Error('Session is disposed'));
+    }
+    let data = JSON.stringify({ type });
```
Comment (minrk) [inline src/session/default.ts]: I did not know that you could use this shorthand for objects!

Comment (blink1073) [inline src/session/default.ts]: Yep, available in ES6.   I am going to also verify that current master of services still works against notebook master (that the old api still works).

**Comment by blink1073** [comment]
Tests pass:

```
Session
[W 09:10:11.490 NotebookApp] Sessions API changed, see updated swagger docs
...
  18 passing (14s)
```

**Comment by minrk** [comment]
Very nice.

**Comment by blink1073** [comment]
Closing as stale.

**Pull request closed without merging**